src/utils/index.js

```diff
@@ -91,22 +91,20 @@ export const blockToDate = (goalBlock, height) => {
   return new Date(diff * 180000 + Date.now());
 };
 
-export const wrapTry = async (promise, store) => {
+// eslint-disable-next-line consistent-return
+export const wrapTry = async (promise) => {
   try {
     return promise.then((res) => {
-      if (!res) {
-        if (store) { store.commit('setBackendStatus', false); }
-        return null;
+      // if (!res) {
+      //   throw new Error('No response object');
+      // }
+      if (res.status !== 200) {
+        throw new Error(`Error code ${res.status}`);
       }
-      if (store) { store.commit('setBackendStatus', true); }
-      if (!res.ok) throw new Error(`Request failed with ${res.status}`);
+      if (!res.ok) throw new Error(`Request failed with status ${res.status}`);
       return res.json();
-    }).catch((error) => {
-      console.error(error);
-      return null;
     });
   } catch (err) {
-    if (store) { store.commit('setBackendStatus', false); }
-    return null;
+    return err;
```

Comment:
who will handle the returned error? Errors should be thrown!

Reply:
I guess that it needs `return throw err` because err is already an instance of Error. will add, thanks

Reply:
I got this lint issue https://eslint.org/docs/rules/no-useless-catch